### PR TITLE
Various

### PR DESCRIPTION
**Pull request by vinniefalco**

### CODE REVIEW

File: include/boost/json/parser.hpp [hunk 1]
```diff
@@ -29,16 +29,17 @@ BOOST_JSON_NS_BEGIN
...
-    must be called, optionally passing the storage
-    pointer to be used by the @ref value container into
-    which the parsed results are stored. After the
-    parse is started, call the @ref write function
-    to provide buffers of characters of the JSON.
-    When there are no more buffers, call @ref finish.
-    The parse is complete when the function @ref done
-    returns `true`, or when a non-successful error
-    code is returned.
+    Parsing for a new JSON may begin the parser is
```
Comment (thats-bonkers) [inline include/boost/json/parser.hpp:32]: This sentence looks broken... possibly meant to say 'when the parser is'

File: include/boost/json/parser.hpp [hunk 2]
```diff
@@ -156,127 +157,137 @@ class parser
...
+        `buffer` extends until the parser is destroyed.
     */
     BOOST_JSON_DECL
     parser(
         storage_ptr sp,
         parse_options const& opt,
-        unsigned char* temp_buffer,
-        std::size_t temp_size) noexcept;
+        unsigned char* buffer,
+        std::size_t size) noexcept;
```
Comment (thats-bonkers) [inline include/boost/json/parser.hpp:193]: Have you considered using a string_view here... or is it just a matter of making calls using `std::byte*` less clunky?

File: include/boost/json/parser.hpp (hunk 1 above)
Comment (vinniefalco) [reply include/boost/json/parser.hpp:32]: This will be fixed elsewhere

File: include/boost/json/parser.hpp (hunk 2 above)
Comment (sdkrystian) [reply include/boost/json/parser.hpp:193]: `string_view` is non-mutable

Comment (thats-bonkers) [reply include/boost/json/parser.hpp:193]: Oh, my bad, I made the wrong assumption here... I guess that a span-like type could be useful here then, but if you don't have any such construct in this project, it doesn't pay off to bring it from another dependency.

File: include/boost/json/parser.hpp (hunk 1 above)
Comment (thats-bonkers) [reply include/boost/json/parser.hpp:32]: Ack

**Review by thats-bonkers — APPROVED** [review]

**Pull request merged**